src/index.test.ts

```diff
@@ -1,27 +1,61 @@
+import { Callback, Context, S3Event, SQSEvent } from 'aws-lambda'
 import { handler, LogRecord } from './index'
 
 import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3'
-import { FirehoseClient, PutRecordCommand } from '@aws-sdk/client-firehose'
+import { FirehoseClient, PutRecordCommand, PutRecordCommandInput } from '@aws-sdk/client-firehose'
 import { mockClient } from 'aws-sdk-client-mock'
 import 'aws-sdk-client-mock-jest'
+import { sdkStreamMixin } from '@smithy/util-stream'
 import { Readable } from 'stream'
 import { promisify } from 'util'
 import { createGunzip, gzip } from 'zlib'
 import { describe } from 'node:test'
-import readline from "node:readline"
+import readline from 'node:readline'
 
 const gzipPromise = promisify(gzip)
 
 const s3Mock = mockClient(S3Client)
 const firehoseMock = mockClient(FirehoseClient)
 
-let testS3EventNotification
+export const mockCallback: Callback = () => undefined
+
+export const mockContext: Context = {
+  awsRequestId: '246fc613-8e0d-482a-9df5-158f2add0665',
+  callbackWaitsForEmptyEventLoop: true,
+  done: () => console.log('Complete'),
+  fail: () => console.log('Error'),
+  functionName: 'firehoseTransform',
+  functionVersion: '$LATEST',
+  getRemainingTimeInMillis: () => 333,
+  invokedFunctionArn: 'arn:aws:lambda:eu-west-2:987654321:function:firehoseTransform',
```

Comment:
Is it from the other lambda? Do we get all these attributes in context for messages from SQS?

Reply:
Oh yeah it is from the other lambda (I got it working in there and copy pasted into here), but you do get this with all these fields (I haven't included the optional ones, only the required ones). This is a lambda context and all lambdas are called with them, the specific arn is a copy paste fail, but the fields are all the required fields in a lambda context.

Although you don't have the argument in your handler, AWS does execute the lambda with one.

You can check out the actual type for the Handler here https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/aws-lambda/handler.d.ts#L84 (and the type for the Context is right below it)

Reply:
Have pushed a commit updating the mockContext so it at least references this lambda :)